### PR TITLE
LogRecord storage at differents levels

### PR DESCRIPTION
**Pull request by jdevillard**
This PR init the capability to choose if we want to store input properties and payload (output) of activity in the Activity Execution Log Record regarding the #4799 

This PR is necessary for the UI development : 
- https://github.com/elsa-workflows/elsa-studio/pull/136
- https://github.com/elsa-workflows/elsa-studio/issues/118

### CODE REVIEW

File: src/clients/Elsa.Api.Client/Resources/WorkflowDefinitions/Enums/PersistenceStrategy.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿namespace Elsa.Api.Client.Resources.WorkflowDefinitions.Enums;
+
+/// <summary>
+/// Define the Persistence strategy to store information
+/// </summary>
+public enum PersistenceStrategy
```
Comment (sfmskywalker) [inline src/clients/Elsa.Api.Client/Resources/WorkflowDefinitions/Enums/PersistenceStrategy.cs:6]: I am wondering if `LogPersistenceMode` might be a better name. It is more focussed at logging, and "mode" is an alternative to "strategy" - we already use "strategy" for types that implement the strategy pattern. "Persistence" on its own is too broad.

What do you think?

**Review by sfmskywalker — CHANGES_REQUESTED** [review]
Looks good, just a small request to revisit the naming of "persistence strategy" etc. to be more specific in order to indicate that this is about persisting **log records**. Otherwise, it might be confusing where we think that this is about persistence in general.

File: src/clients/Elsa.Api.Client/Resources/WorkflowDefinitions/Enums/PersistenceStrategy.cs (hunk 1 above)
Comment (jdevillard) [reply src/clients/Elsa.Api.Client/Resources/WorkflowDefinitions/Enums/PersistenceStrategy.cs:6]: Yes , agree ! I'll do the change

Comment (jdevillard) [reply src/clients/Elsa.Api.Client/Resources/WorkflowDefinitions/Enums/PersistenceStrategy.cs]: Change done , using the two commit, I also rename the json key. 
I'll update the Client code once this PR will be done.

**Comment by sfmskywalker** [comment]
@jdevillard We will merge this once you give us the green light!

**Comment by jdevillard** [comment]
Ok I found the issue about yesterday demo , 

it was one case when you revert Workflow Configuration to "Default" so the value of the property was LogPersistenceMode.Default and not LogPersistenceMode.Include or LogPersistenceMode.Exclude so the information was loosed when the algo have to record or not the data.

So now I think It's ok  for me !

thx a lot !

**Review by sfmskywalker — APPROVED** [review]

**Comment by sfmskywalker** [comment]
Awesome, great job @jdevillard . Many thanks for this!

**Pull request merged**